src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -106,8 +106,9 @@ public HttpApiHandler(DurableTaskExtension config, ILogger logger)
             if (i < 0)
             {
                 // Retrive All Status in case of the request URL ends e.g. /instances/
+                int indexOfSegment = request.RequestUri.AbsolutePath.TrimEnd('/').IndexOf(InstancesControllerSegment.TrimEnd('/'), StringComparison.OrdinalIgnoreCase);
                 if (request.Method == HttpMethod.Get
-                    && path.IndexOf(InstancesControllerSegment.TrimEnd('/'), StringComparison.OrdinalIgnoreCase) >= 0)
+                    && string.Equals(path?.Substring(indexOfSegment), InstancesControllerSegment.TrimEnd('/'), StringComparison.OrdinalIgnoreCase))

```

Comment:
Isn't it simpler to just say `string.Equals(path, InstancesControllerSegment.TrimEnd('/'), StringComparison.OrdinalIgnoreCase)`? I don't think you need the `indexOfSegment` variable.

Reply:
Why we need this is, path is not "/Instances/". Actually it is "/admin/extensions/DurableTaskExtension/Instances". Since the InstancesControllerSegment is "/Instances/" we need to do Substring it.

Reply:
It is much cooler than my code. :) Thanks! Done